pulseapi/profiles/models/categories.py

```diff
@@ -59,3 +63,80 @@ class ProgramYear(models.Model):
 
     def __str__(self):
         return self.value
+
+
+class UserProfileProgramParticipation(models.Model):
+    profile = models.ForeignKey(
+        'profiles.UserProfile',
+        on_delete=models.CASCADE,
+        related_name='program_participation',
+    )
+
+    program = models.ForeignKey(
+        'profiles.ProgramType',
+        on_delete=models.CASCADE,
+        related_name='participation',
+    )
+
+    year = models.PositiveSmallIntegerField(
+        # TODO: Change to MaxValueValidator with callable when
+        #       we update to Django > v2.2
+        validators=[YearValidator(max_offset=2)],
+        null=True,
+        blank=True,
+    )
+
+    cohort = models.CharField(
+        null=True,
+        blank=True,
+        max_length=200,
+    )
+
+    def __str__(self):
+        return f'{self.profile.name} - {self.program} {self.year} {self.cohort}'
+
+    def clean(self):
+        # Don't allow both the cohort and year to be empty
+        if self.year is None and not self.cohort:
+            raise ValidationError(
+                _('Either the year or cohort must have a value')
+            )
+
+    class Meta:
+        verbose_name = 'Profile participation in a program'
```

Comment:
hm, what would the plural be?

Reply:
ah good question I'll add that in as "List of profile/program associations" or something like that